index.js

```diff
@@ -212,24 +241,58 @@ NYC.prototype.report = function (cb, _collector, _reporter) {
   reporter.write(collector, true, cb)
 }
 
+var loadedMaps = {}
+
 NYC.prototype._loadReports = function () {
   var _this = this
   var files = fs.readdirSync(this.tmpDirectory())
 
-  return _.map(files, function (f) {
+  var SourceMapCache = require('./lib/source-map-cache')
+  var sourceMapCache = new SourceMapCache()
+
+  var cacheDir = _this.cacheDirectory()
+
+  return files.map(function (f) {
+    var report
     try {
-      return JSON.parse(fs.readFileSync(
+      report = JSON.parse(fs.readFileSync(
         path.resolve(_this.tmpDirectory(), './', f),
         'utf-8'
       ))
     } catch (e) { // handle corrupt JSON output.
       return {}
     }
+
+    if (report) {
+      Object.keys(report).forEach(function (relFile) {
+        var fileReport = report[relFile]
+        if (fileReport && fileReport.contentHash) {
+          var hash = fileReport.contentHash
+          if (!(loadedMaps[hash] || (loadedMaps[hash] === false))) {
+            try {
+              var mapPath = path.join(cacheDir, hash + '.map')
+              loadedMaps[hash] = fs.readFileSync(mapPath, 'utf8')
+            } catch (e) {
+              loadedMaps[hash] = false
+            }
+          }
+          if (loadedMaps[hash]) {
+            sourceMapCache.addMap(relFile, loadedMaps[hash])
+          }
+        }
+      })
+      report = sourceMapCache.applySourceMaps(report)
+    }
+    return report
   })
 }
 
-NYC.prototype.tmpDirectory = function () {
-  return path.resolve(this.cwd, './', this.tempDirectory)
+NYC.prototype.tmpDirectory = NYC.prototype.tempDirectory = function () {
```

Comment:
`tmpDirectory` is there for backwards compatibility. I don't think anybody is using that outside us, and it certainly is not documented. I think it's safe just to drop the other one.